### ISSUE TITLE
Workflow `submodules: recursive` setting not detected by action

### ISSUE THREAD

**Issue by supleed2** [actions/checkout#1443]
The `actions/checkout@v3` workflow doesn't seem to respect the `with: submodules: recursive` setting.

In [this](https://github.com/supleed2/omg-rs/actions/runs/6027738760/workflow) workflow, the file shows the following:

```yaml
jobs:
  create-release:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
        with:
          submodules: recursive
```

but in the logs (located [here](https://github.com/supleed2/omg-rs/actions/runs/6027738760/job/16353579296#step:2:12)) it shows `false`:

```yaml
Run actions/checkout@v3
  with:
    repository: supleed[2](https://github.com/supleed2/omg-rs/actions/runs/6027738760/job/16353579296#step:2:2)/omg-rs
    token: ***
    ssh-strict: true
    persist-credentials: true
    clean: true
    sparse-checkout-cone-mode: true
    fetch-depth: 1
    fetch-tags: false
    lfs: false
    submodules: false
    set-safe-directory: true
```

so the following build fails. Am I configuring something wrong?

**Comment by allanrogerr** [comment]
This is still happening
